# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/windows/PCompositorWidget.ipdl
# Commit: 71e63781b38c
# Full Hash: 71e63781b38cbde024a9057553cae96c98e5ecb2
# Author: Chris Martin <cmartin@mozilla.com>
# Date: 2020-02-13 18:21:33
# Regressor Bug: 1604412
# File Overlap Count: 2
# Description:
#   Bug 1604412 - Enable remote backbuffer GDI compositing r=jmathies,jld
#   
#   This change adds new "remote backbuffer" logic when compositing without
#   HW acceleration on Windows (IE compositing through Cairo using the Win32
#   GDI)
# ==============================================================================

diff -r f2088b5ae50d -r 71e63781b38c widget/windows/PCompositorWidget.ipdl
--- a/widget/windows/PCompositorWidget.ipdl	Thu Feb 13 07:13:50 2020 +0200
+++ b/widget/windows/PCompositorWidget.ipdl	Thu Feb 13 03:59:13 2020 +0000
@@ -13,11 +13,19 @@
 namespace mozilla {
 namespace widget {
 
+struct RemoteBackbufferHandles {
+  WindowsHandle fileMapping;
+  WindowsHandle requestReadyEvent;
+  WindowsHandle responseReadyEvent;
+};
+
 sync protocol PCompositorWidget
 {
   manager PCompositorBridge;
 
 parent:
+  sync Initialize(RemoteBackbufferHandles aRemoteHandles);
+  
   sync EnterPresentLock();
   sync LeavePresentLock();
   async UpdateTransparency(nsTransparencyMode aMode);